src/main/java/tasklist/TaskList.java

```diff
@@ -0,0 +1,193 @@
+package tasklist;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import command.CommandNotFoundException;
+import ouiouibaguette.OuiOuiBaguetteException;
+import storage.Storage;
+import tasks.Deadline;
+import tasks.Event;
+import tasks.Task;
+import tasks.ToDo;
+
+/**
+ * Represents a list of tasks and provides methods to manipulate and access these tasks.
+ * The tasks are loaded from and stored in a given storage system.
+ */
+public class TaskList {
+
+    /** The storage system used to persist the tasks. */
+    protected Storage storage;
+
+    /** The list that holds all the tasks. */
+    protected List<Task> tasklist;
+
+    /**
+     * Constructs a TaskList with the specified storage system.
+     * Loads the tasks from the storage into the task list.
+     *
+     * @param storage The storage system used to persist the tasks.
+     */
+    public TaskList(Storage storage) {
+        // Check if storage is null
+        assert storage != null : "Storage argument cannot be null";
+
+        this.storage = storage;
+        tasklist = new ArrayList<Task>();
+
+        // Load data from storage
+        List<String> data = storage.load();
+
+        for (String line : data) {
```

Comment:
Would it be better rename line to taskEntry? This might improve clarity by making the purpose of the variable more evident.